mpas_analysis/ocean/nino34_timeseries.py

```diff
@@ -0,0 +1,198 @@
+    their mpas_analysis counterparts.
+
+    If present, variableMap is a dictionary of MPAS-O variable names that map
+    to their mpas_analysis counterparts.
+
+    Author: Luke Van Roekel
+    Last Modified: 03/21/2017
+    """
+
+    # Define/read in general variables
+    print '  Load SST data...'
+    # read parameters from config file
+    inDirectory = config.get('input', 'baseDirectory')
+
+    streamsFileName = config.get('input', 'oceanStreamsFileName')
+    streams = StreamsFile(streamsFileName, streamsdir=inDirectory)
+
+    namelistFileName = config.get('input', 'oceanNamelistFileName')
+    namelist = NameList(namelistFileName, path=inDirectory)
+
+    calendar = namelist.get('config_calendar_type')
+    simulationStartTime = get_simulation_start_time(streams)
+
+    # get a list of timeSeriesStats output files from the streams file,
+    # reading only those that are between the start and end dates
+    startDate = config.get('timeSeries', 'startDate')
+    endDate = config.get('timeSeries', 'endDate')
+    streamName = streams.find_stream(streamMap['timeSeriesStats'])
+    fileNames = streams.readpath(streamName, startDate=startDate,
+                                 endDate=endDate,  calendar=calendar)
+    print 'Reading files {} through {}'.format(fileNames[0], fileNames[-1])
+
+    mainRunName = config.get('runs', 'mainRunName')
+    plotsDirectory = buildConfigFullPath(config, 'output', 'plotsSubdirectory')
+
+    plotTitles = config.getExpression('regions', 'plotTitles')
+#   regionIndex should correspond to NINO34 in surface weighted Average AM
+    regionIndex = config.getint('timeSeriesNino34', 'regionIndex')
+
+#   Load data:
+
+    varList = ['avgSurfaceTemperature']
+    ds = open_multifile_dataset(fileNames=fileNames,
+                                calendar=calendar,
+                                simulationStartTime=simulationStartTime,
+                                timeVariableName='Time',
+                                variableList=varList,
+                                variableMap=variableMap,
+                                startDate=startDate,
+                                endDate=endDate)
+
+    SSTregions = ds.avgSurfaceTemperature
+    print '  Compute NINO3.4 index...'
+    nino34 = compute_nino34_index(SSTregions[:, regionIndex], config)
+
+    print ' Computing NINO3.4 power spectra...'
+#   NOTE: first 2 and last two values ommitted due to rolling mean
+    f, spectra, conf99, conf95, redNoise = compute_nino34_spectra(nino34, config)
+
+    start = days_to_datetime(np.amin(ds.Time.min()), calendar=calendar)
+    end = days_to_datetime(np.amax(ds.Time.max()), calendar=calendar)
+
+#   Convert frequencies to period in years
+    f = 1.0 / (1E-9 + f*86400.*365)
+    print ' Plot NINO3.4 index and spectra...'
+    xLabel = 'Time [years]'
+    yLabel = '[$^\circ$ C]'
+
+    figureName = '{}/NINO34_{}.png'.format(plotsDirectory, mainRunName)
```

Comment:
I'm unsure what you are requesting here.  Do you want me to change something in the figure name?